File: src/main/java/cn/chahuyun/economy/utils/Log.java
package cn.chahuyun.economy.utils;

import cn.chahuyun.economy.HuYanEconomy;
import net.mamoe.mirai.utils.MiraiLogger;

/**
 * 日志<p>
 * 使用此类日志做统一处理
 *
 * @author Moyuyanli
 * @date 2022/11/14 13:01
 */
public class Log {

    private static final MiraiLogger log = HuYanEconomy.INSTANCE.getLogger();

    private static String name = "壶言经济--";

    private Log() {

    }

    /**
     * 提示级信息
     *
     * @param msg 日志信息
     * @author Moyuyanli
     * @date 2022/11/14 13:04
     */
    public static void info(String msg) {
        log.info(name + msg);
    }


    /**
     * 警告级信息
     *
     * @param msg 日志信息
     * @author Moyuyanli
     * @date 2022/11/14 13:23
     */
    public static void warning(String msg) {
        log.warning(name + msg);
    }

    /**
     * 错误级信息
     *
     * @param msg 消息
     * @author Moyuyanli
     * @date 2022/11/14 13:25
     */
    public static void error(String msg) {
        log.error(name + msg);
    }

    /**
     * 错误级信息
     *
     * @param e 异常
     * @author Moyuyanli
     * @date 2022/11/14 15:20
     */

    public static void error(Throwable e) {
        log.error(name + e.getMessage(), e);
    }

    /**
     * 错误级信息
     *
     * @param msg 消息
     * @param e   异常
     * @author Moyuyanli
     * @date 2022/11/14 15:19
     */
    public static void error(String msg, Throwable e) {
        log.error(name + msg);
        log.error(name + e.getMessage(), e);
    }

    /**
     * 调试级信息
     *
     * @param msg 消息
     * @author Moyuyanli
     * @date 2022/11/14 14:44
     */
    public static void debug(String msg) {
        log.debug(name + msg);
    }

    /**
     * 调试级信息
     *
     * @param exception 异常
     * @author Moyuyanli
     * @date 2022/12/10 0:55
     */
    public static void debug(Exception exception) {
        log.debug(name + exception);
    }


    public static void setName(String name) {
        Log.name = name;
    }
}

File: src/main/kotlin/cn/chahuyun/economy/utils/Log.kt
package cn.chahuyun.economy.utils

/**
 * Common logger facade.
 *
 * Modules outside economy-main must not know about the Mirai plugin instance.
 * The plugin entry wires the real Mirai logger during startup.
 */
object Log {
    private var sink: Sink = ConsoleSink

    @JvmField
    var name: String = "壶言经济--"

    interface Sink {
        fun info(message: String)
        fun warning(message: String)
        fun error(message: String)
        fun error(message: String, throwable: Throwable)
        fun debug(message: String)
    }

    private object ConsoleSink : Sink {
        override fun info(message: String) = println(message)
        override fun warning(message: String) = println(message)
        override fun error(message: String) = System.err.println(message)
        override fun error(message: String, throwable: Throwable) {
            System.err.println(message)
            throwable.printStackTrace()
        }

        override fun debug(message: String) = println(message)
    }

    @JvmStatic
    fun configure(sink: Sink) {
        this.sink = sink
    }

    @JvmStatic
    fun configure(
        info: (String) -> Unit,
        warning: (String) -> Unit,
        error: (String) -> Unit,
        errorWithThrowable: (String, Throwable) -> Unit,
        debug: (String) -> Unit,
    ) {
        sink = object : Sink {
            override fun info(message: String) = info(message)
            override fun warning(message: String) = warning(message)
            override fun error(message: String) = error(message)
            override fun error(message: String, throwable: Throwable) = errorWithThrowable(message, throwable)
            override fun debug(message: String) = debug(message)
        }
    }

    @JvmStatic
    fun info(msg: String) {
        sink.info(name + msg)
    }

    @JvmStatic
    fun warning(msg: String) {
        sink.warning(name + msg)
    }

    @JvmStatic
    fun error(msg: String) {
        sink.error(name + msg)
    }

    @JvmStatic
    fun error(e: Throwable) {
        sink.error(name + (e.message ?: "unknown error"), e)
    }

    @JvmStatic
    fun error(msg: String, e: Throwable) {
        sink.error(name + msg)
        sink.error(name + (e.message ?: "unknown error"), e)
    }

    @JvmStatic
    fun debug(msg: String) {
        sink.debug(name + msg)
    }

    @JvmStatic
    fun debug(exception: Exception) {
        sink.debug(name + exception)
    }
}
